.ci/Jenkinsfile

```diff
@@ -178,6 +179,35 @@ pipeline {
           githubNotify(context: "${env.GITHUB_CHECK_ITS_NAME}", description: "${env.GITHUB_CHECK_ITS_NAME} ...", status: 'PENDING', targetUrl: "${env.JENKINS_URL}search/?q=${env.ITS_PIPELINE.replaceAll('/','+')}")
         }
       }
+      stage('Release') {
+        agent { label 'linux && immutable' }
+        options { skipDefaultCheckout() }
+        environment {
+          HOME = "${env.WORKSPACE}"
+          PATH = "${env.WORKSPACE}/bin:${env.PATH}"
+          RUBY_DOCKER_TAG = 'ruby:latest'
+        }
+        when {
+          beforeAgent true
+          anyOf {
+            tag pattern: 'v\\d+.*', comparator: 'REGEXP'
+            expression { return params.Run_As_Master_Branch }
+          }
+        }
+        steps {
+          withGithubNotify(context: 'Release') {
+            deleteDir()
+            unstash 'source'
+            script {
+              sshagent(['f6c7695a-671e-4f4f-a331-acdce44ff9ba']) {
```

Comment:
I am not sure if the key will be in the Docker container

Reply:
I'd go for the docker.inside + sshagent then